src/main/java/edu/greenblitz/robotName/subsystems/LED/LED.java

```diff
@@ -0,0 +1,165 @@
+		this.addressableLEDBuffer = new AddressableLEDBuffer(LED_LENGTH);
+		this.addressableLED.setLength(LED_LENGTH);
+		this.addressableLED.start();
+		
+		LEDBlinkTimer = new Timer();
+		actionTimer = new Timer();
+		LEDBlinkTimer.restart();
+		
+		currentColor = SPEAKER_MODE_COLOR;
+		wasNoteInRobot = false;
+
+		chooser = new SendableChooser<>();
+		chooser.setDefaultOption("False", false);
+		chooser.addOption("True", true);
+		SmartDashboard.putData("Is npot robot", chooser);
+	}
+	
+	public static LED getInstance() {
+		init();
+		return instance;
+	}
+	
+	public static void init() {
+		if (instance == null) {
+			instance = new LED();
+		}
+	}
+	
+	public boolean getWasNoteInRobot(){
+		return wasNoteInRobot;
+	}
+	
+	public void setLEDColor(Color color, Section section) {
+		setLEDColor(color, section.startIndex(), section.endIndex());
+	}
+	
+	public void setLEDColor(Color color, int startIndex, int endIndex) {
+		for (int i = startIndex; i < endIndex; i++) {
+			setLEDColor(color, i);
+		}
+	}
+	
+	public void setLEDColor(Color color, int index) {
+		this.addressableLEDBuffer.setLED(index, color);
+	}
+	
+	public void turnOff(int index) {
+		setLEDColor(new Color(0, 0, 0), index);
+	}
+	
+	public void turnOff(int startIndex, int endIndex) {
+		for (int i = startIndex; i < endIndex; i++) {
+			turnOff(i);
+		}
+	}
+	
+	public void turnOff(Section section) {
+		turnOff(section.startIndex(), section.endIndex());
+	}
+	
+	@Override
+	public void periodic() {
+		currentColor = getColorByMode();
+		this.addressableLED.setData(addressableLEDBuffer);
+	}
+	
+	public void setColorByMode() {
+		setLEDColor(currentColor, LEDConstants.ALL_LED);
+	}
+	
+	public Color getColorByMode() {
+		if (ScoringModeSelector.isAmpMode()) {
+			return LEDConstants.AMP_MODE_COLOR;
+		} else {
+			return LEDConstants.SPEAKER_MODE_COLOR;
+		}
+	}
+
+	public boolean isNoteInRobot() {
+//		return (Intake.getInstance().getExitBeamBreakerValue()
+//				|| Intake.getInstance().getEntranceBeamBreakerValue()
+//				|| Funnel.getInstance().isObjectIn()
+//				|| Roller.getInstance().isObjectInside());
+		return chooser.getSelected();
+	}
+
+	public void updateNoteState(){
+		wasNoteInRobot = isNoteInRobot();
+	}
+	
+	public void blink(Color color) {
+		if ((LEDBlinkTimer.get()) % (LEDConstants.BLINK_DURATION * 2) >= LEDConstants.BLINK_DURATION) {
+			LED.getInstance().turnOff(LEDConstants.ALL_LED);
+		} else {
+			LED.getInstance().setLEDColor(color, LEDConstants.ALL_LED);
+		}
+	}
+	
+	public void rumble(){
+		OI.getInstance().getMainJoystick().rumble(LEDConstants.RUMBLE_LEFT_MOTOR, LEDConstants.RUMBLE_POWER);
+	}
+	
+	public void stopRumble(){
+		OI.getInstance().getMainJoystick().rumble(LEDConstants.RUMBLE_LEFT_MOTOR, 0);
+	}
+	
+	public void restartTimerByNoteState(){
+		if (getWasNoteInRobot() && !isNoteInRobot()) {
+			actionTimer.restart();
+		}
+		else if (!getWasNoteInRobot() && isNoteInRobot()) {
+			actionTimer.restart();
+		}
```

Comment:
romy - douplecate code..
```suggestion
		if (getWasNoteInRobot() && !isNoteInRobot() 
		    || !getWasNoteInRobot() && isNoteInRobot()) {
			actionTimer.restart();
		}
```

Reply:
> romy - douplecate code..
no this is stupid.
this is how you do it
```suggestion
		if (getWasNoteInRobot() != isNoteInRobot()) {
			actionTimer.restart();
		}
```